CMakeLists.txt

```diff
@@ -117,25 +118,30 @@ target_link_libraries(
         $<$<STREQUAL:${MLN_WITH_SANITIZER},undefined>:-fsanitize=undefined>
 )
 
-cmake_policy(SET CMP0063 NEW)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_C_EXTENSIONS OFF)
 set(CMAKE_C_STANDARD 11)
 set(CMAKE_C_STANDARD_REQUIRED ON)
-set(CMAKE_C_VISIBILITY_PRESET hidden)
 if (CMAKE_GENERATOR STREQUAL "Ninja Multi-Config")
     set(CMAKE_CROSS_CONFIGS "all")
     set(CMAKE_DEFAULT_CONFIGS "all")
 endif()
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_MINSIZEREL ON)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
 
-if(MLN_WITH_QT)
+if (NOT MLN_WITH_EMSCRIPTEN)
```

Comment:
Naturally this causes the size of the wasm file to skyrocket: 40 MB. Once I get the bloody thing working I'll hopefully manage to do something about this.